app/models/manageiq/providers/openstack/cloud_manager/cloud_volume/operations.rb

```diff
@@ -8,6 +8,7 @@ def validate_detach_volume
   end
 
   def raw_attach_volume(server_ems_ref, device = nil)
+    device = nil if (device && device.empty?)
```

Comment:
I'll wait for further feedback before fixing this -- let me know which style to use. 
Between these two, I'd prefer the second. When using && and || I prefer explicit parens since I've been bitten by '&&/||' vs. 'and/or' operator precedence bugs in the past. If parens are discouraged here, the second avoids even the appearance of ambiguity.